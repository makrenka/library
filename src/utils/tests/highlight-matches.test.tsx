import { describe, it } from "@jest/globals";
import renderer from 'react-test-renderer';
import { highlightMatches } from "../highlight-matches";
import React from "react";

describe('highlight matches', () => {
    it('returns str', () => {
        expect(highlightMatches('', 'хулиномика')).toBe('хулиномика');
    })

    it('searches matches', () => {
        const substr = "номика";
        const component = renderer
            .create(
                <React.Fragment key={substr}>
                    {substr}
                    <span style={{ color: '#FF5253' }} data-test-id='highlight-matches'>
                        хули
                    </span>
                </React.Fragment>
            ).toJSON();

        expect(component).toMatchSnapshot();
    });
});