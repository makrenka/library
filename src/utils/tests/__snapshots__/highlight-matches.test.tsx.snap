// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`highlight matches searches matches 1`] = `
[
  "номика",
  <span
    data-test-id="highlight-matches"
    style={
      {
        "color": "#FF5253",
      }
    }
  >
    хули
  </span>,
]
`;
